### DISCUSSION TITLE
dmp-file size tasmota13.1.0 seems to mismatch decode-config file size requirement

### DISCUSSION THREAD

**Discussion by cruiser2** [tasmota/decode-config#71] [Q&A] [answered]
Dmp-file size on 13.1.0 = 4.560 Bytes. Decode-config.exe for 13.1.0 expects 4.096 Bytes.
Tasmota developer confirmed increased file size as by design.

**Comment by curzon01** [comment]
if so, that's an issue (#72) - will clarify more details, don't want to cut the additonal data if any

**Comment by curzon01** [comment]
Try current dev version. It's a workaround, not final solution (will try to enhance so driver setting files are included in backup/restore)

**Comment by curzon01** [comment, marked as answer]
[v13.1.0.1 [1022947]](https://github.com/tasmota/decode-config/tree/b367e725772181b44d150289adde22dfd56d77f0) now supports driver config backup and restore.

The key in [JSON file](https://github.com/tasmota/decode-config#json-format-1) is `"settings"` and contains the base64 encoded binary data for each driver name e. g.
```json
"settings": {
    "/.drvset003": "IzrFUQIB8gCMAAAAAAADBAAAAAAAAAAAAAAAAAAAAADyMAAA8jAAAPIwAADyMAAA8jAAAPIwAADyMAAA8jAAAJ4HAACeBwAAngcAAJ4HAACeBwAAngcAAJ4HAACeBwAArA0AAKwNAACsDQAArA0AAKwNAACsDQAArA0AAKwNAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACPwvU9KVwPPgrXIz4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAUAAAAAAAAAAAAAAAAAAAAKQBeACwBHgAAAAAAAAAAAAAAAAAPQrXPgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA9Ctc+", 
    "/.drvset122": "2QXHWgEBAABUZXh0MV9fX18AVGV4dDJfX19fACMAAAAAAAAAAABUZXh0NF9fX18AIwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFRleHQxNV9fXwBUZXh0MTYAAAAA"
},
```

For the [dmp file](https://github.com/tasmota/decode-config#dmp-format) backup the additonal data is appended like Tasmota does it.

The [bin file](https://github.com/tasmota/decode-config#bin-format) is the same as dmp file but unscrambled including attached config files and still has the additional bin file magic value attached.

**Reply by poggenpower** [reply]
@curzon01  I have tested v13.1.0.1 and out in json is working fine:
`decode-config.py -s http://esp04.wupp`
But getting this as commands, fail:
```
> decode-config.py -s http://esp04.wupp -T cmnd
WARNING 9: Restore file contains obsolete name 'settings', skipped
Premature exit - #9 Restore data error
```
I am running 13.1.0 on esp32 (self compiled).
`decode-config.py -s http://esp04.wupp -T cmnd -w` helped to get an output, but I thought `-w` is only required if I convert to another version.

**Reply by curzon01** [reply]
will check it also with latest v13.1.0.2...

**Reply by curzon01** [reply]
forgot these exceptions, use latest dev...
